packages/eas-cli/src/credentials/ios/appstore/bundleIdCapabilities.ts

```diff
@@ -156,7 +167,23 @@ function getCapabilitiesToEnable(
     // Only skip if the existing capability is a simple boolean value,
     // if it has more complex settings then we should always update it.
     // If the `existing.attributes.settings` object is defined, then we can determine that it has extra configuration.
-    if (existing && existing?.attributes.settings == null) {
+    // For push notifications, we should always update the capabaility if
+    // - settings are not defined in the existing capability, but usesBroadcastPushNotifications is enabled (we want to add settings for this capability)
+    // - settings are defined in the existing capability, but usesBroadcastPushNotifications is disabled (we want to remove settings for this capability)
+    const isPushNotificationsCapability =
+      staticCapabilityInfo.capability === CapabilityType.PUSH_NOTIFICATIONS;
+    const noSettingsAttributes = existing?.attributes.settings == null;
+    const hasBroadcastPushNotificationsDisabledBothLocalAndRemote =
+      !additionalOptions.usesBroadcastPushNotifications && noSettingsAttributes;
+    const hasBroadcastPushNotificationsEnabledBothLocalAndRemote =
+      additionalOptions.usesBroadcastPushNotifications && !noSettingsAttributes;
```

Comment:
Can we replace it with `const isPushNotificationSameLocalAndRemove = additionalOptions.usesBroadcastPushNotifications === noSettings`? Do you think it would be clearer?